lib/puppet/provider/rabbitmq_queue/rabbitmqadmin.rb

```diff
@@ -31,7 +31,7 @@ def self.instances
         if arguments.nil?
           arguments = '{}'
         else
-          arguments = arguments.gsub(%r{^\[(.*)\]$}, '').gsub(%r{\{("(?:.|\\")*?"),}, '{\1:').gsub(%r{\},\{}, ',')
+          arguments = arguments.gsub(%r{^\[|\]$}, '').gsub(%r{\{("(?:.|\\")*?"),}, '{\1:').gsub(%r{\},\{}, ',')
```

Comment:
The unit test for this one defines a queue with no arguments, and then has the stubbed shell call return a second instance _with_ arguments

Maybe this could be adjusted slightly, to make sure that the arguments are parsed properly, for example, by adding a check similar to the one you added in the new unit test?
https://github.com/voxpupuli/puppet-rabbitmq/blob/83aa697583224d5761f922b5fe710d9faa3f7a46/spec/unit/puppet/provider/rabbitmq_queue/rabbitmqadmin_spec.rb#L23